File: 1940-maximum-xor-for-each-query/maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        max_num = pow(2, maximumBit) - 1
        for i in range(1, len(nums)):
            nums[i] ^= nums[i - 1]
        for i in range(len(nums)):
            nums[i] ^= max_num
        nums.reverse()
        return nums


        # for i in range(len(nums)):
        #     temp = nums[0]
        #     for j in range(1, len(nums) - i):
        #         temp ^= nums[j]
        #     answer.append(temp)
        # print(answer)
        # return answer

File: 1940-maximum-xor-for-each-query/maximum-xor-for-each-query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int xors = 0;
        for (int i: nums)
            xors ^= i;
        
        int maxi = (1 << maximumBit) - 1;

        int[] result = new int[nums.length];
        int j = 0;
        for (int i = nums.length - 1; i >= 0; i--){
            result[j] = xors ^ maxi;
            xors ^= nums[i];
            j += 1;
        }
        return result;
    }
}